File: src/programmers/level1/하샤드.py
def solution(x):
    answer = True
    x1 = x
    sum = 0
    while(True):
      sum+= x1 % 10
      x1 = x1//10
      if x1<1:
        break
    if x%sum!=0: answer = False
    return answer

result = solution(12)
print(result)

File: src/programmers/level1/하샤드.js
function fun(sum, x){
  console.log(sum,x);

  if(x < 10) {
    return sum+x;
  }
  return fun(sum+x%10, parseInt(x/10)) ;
}
function solution(x) { 
  if(x % fun(0, x) == 0) return true;
  else return false;
}

console.log('answer : ' + solution(10));


/*
function Harshad(n){
  return !(n % (n + "").split("").reduce((a, b) => +b + +a ));
}
*/